megamek/src/megamek/common/actions/WeaponAttackAction.java

```diff
@@ -5057,12 +5070,40 @@ private static ToHitData handleArtilleryAttacks(Game game, Entity ae, Targetable
 
         // Handle direct artillery attacks.
         if (isArtilleryDirect) {
+            // Air-Defense Arrow missiles use a much-simplified to-hit calculation because they:
+            // A) are Flak; B) are not Artillery, C) use three ranges (same low-alt hex, 1 hex, 2 hexes)
+            // as S/M/L
+            // Per TO:AR 6th printing, p153, other mods are: Flak (-2), AMM, damage, intervening trees/jungle
+            if (atype != null && atype.getMunitionType().contains(AmmoType.Munitions.M_ADA)){
```

Comment:
I'm thinking we may consider moving the internal contents of this if block to a separate method to avoid adding to this already massive method.

Reply:
I'll go ahead and split both direct and indirect to-hit calcs out into smaller functions.